File: javascript/jssearch/tests/filetypes.test.js
/*
 * filetypes_test.js
 *
 * Some tests of filetypes.js
 */

const {FileType} = require('../src/filetype');
const {FileTypes} = require('../src/filetypes');

describe('testing filetypes', () => {
    it('testFileTypesArchiveFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'archive.zip';
        const res = fileTypes.isArchiveFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.ARCHIVE);

        fileTypes.getFileTypeAsync(filename, (err, type) => {
            expect(err).toBeNull();
            expect(type).toBe(FileType.ARCHIVE);
        });

        const invalid_filename = 200;
        fileTypes.getFileTypeAsync(invalid_filename, (err, t) => {
            // console.log('err: ' + err);
            expect(err).toBeDefined();
            // console.log('type: ' + t);
            expect(t).toBeUndefined();
        });
    });

    it('testFileTypesBinaryFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'binary.exe';
        const res = fileTypes.isBinaryFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.BINARY);
    });

    it('testFileTypesCodeFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'code.js';
        const res = fileTypes.isCodeFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.CODE);
    });

    it('testFileTypesTextFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'text.txt';
        const res = fileTypes.isTextFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.TEXT);
    });

    it('testFileTypesXmlFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'markup.xml';
        const res = fileTypes.isXmlFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.XML);
    });

    it('testFileTypesSearchableFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'README.md';
        const res = fileTypes.isSearchableFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.TEXT);
    });

    it('testFileTypesUnknownFile', () => {
        const fileTypes = new FileTypes();
        const filename = 'unknown.xyz';
        const res = fileTypes.isUnknownFile(filename);
        expect(res).toBeTruthy();
        const type = fileTypes.getFileType(filename);
        expect(type).toBe(FileType.UNKNOWN);
    });
});

File: dart/dartsearch/test/file_types_test.dart
import 'package:test/test.dart';

void main() {
  var fileTypes = FileTypes();

  test('test archive file', () async {
    var fileName = 'archive.zip';
    expect(await fileTypes.isArchiveFile(fileName), true);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isSearchableFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.archive);
  });

  test('test binary file', () async {
    var fileName = 'binary.exe';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), true);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isSearchableFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.binary);
  });

  test('test code file', () async {
    var fileName = 'code.dart';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), true);
    expect(await fileTypes.isSearchableFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.code);
  });

  test('test text file', () async {
    var fileName = 'README.md';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isSearchableFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.text);
  });

  test('test xml file', () async {
    var fileName = 'markup.xml';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isSearchableFile(fileName), true);
    expect(await fileTypes.isTextFile(fileName), true);
    expect(await fileTypes.isXmlFile(fileName), true);
    expect(await fileTypes.getFileType(fileName), FileType.xml);
  });

  test('test unknown file', () async {
    var fileName = 'unknown.ZZZ';
    expect(await fileTypes.isArchiveFile(fileName), false);
    expect(await fileTypes.isBinaryFile(fileName), false);
    expect(await fileTypes.isCodeFile(fileName), false);
    expect(await fileTypes.isSearchableFile(fileName), false);
    expect(await fileTypes.isTextFile(fileName), false);
    expect(await fileTypes.isXmlFile(fileName), false);
    expect(await fileTypes.getFileType(fileName), FileType.unknown);
  });
}
